cetmix_tower_server/models/cetmix_tower.py

```diff
@@ -99,3 +102,50 @@ def server_get_variable_value(
         if not value and check_global:
             value = result.get("global")
         return value
+
+    @api.model
+    def server_check_ssh_connection(self, server_reference, attempts=5, timeout=15):
+        """Check if SSH connection to the server is available.
+
+        Args:
+            server_reference (Char): Server reference
+            attempts (int): Number of attempts to try the connection.
+                Default is 5.
+            timeout (int): Timeout in seconds for each connection attempt.
+                Default is 15 seconds.
+
+        Returns:
+            dict: {
+                "code": int,  # 0 for success, -1 for not found,
+                 1 for connection error, 2 for retry limit exceeded
+                "message": str  # Description of the result
+            }
+        """
+        server = self.env["cx.tower.server"].get_by_reference(server_reference)
+        if not server:
+            return {"code": -1, "message": _("Server not found")}
+
+        ssh_host = server.ip_v4_address or server.ip_v6_address
+        ssh_port = server.ssh_port if server.ssh_port else 22
+
+        for attempt in range(attempts):
+            try:
+                # Create an SSH client
+                ssh_client = paramiko.SSHClient()
```

Comment:
You should use existing [_connect](https://github.com/cetmix/cetmix-tower/blob/75ef96fc0a4224968b07ba30bf3a690d00598ee6/cetmix_tower_server/models/cx_tower_server.py#L106) method of the SSH class instead.